symengine/complex.h

```diff
@@ -185,17 +186,24 @@ class Complex : public Number
      * */
     inline RCP<const Number> divcomp(const Complex &other) const
     {
-        rational_class modulus_sq
+        rational_class modulus_sq_this
+            = this->real_ * this->real_ + this->imaginary_ * this->imaginary_;
```

Comment:
Can you move this inside, `if (get_num(modulus_sq_other) == 0) {` block? (This will probably be done by the compiler itself, but move this just to be sure.)